main/swadge2024.c

```diff
@@ -402,40 +401,31 @@ void app_main(void)
 
                 if (tHeldUs > PAUSE_TIME_US)
                 {
-                    if (NULL != overlayMode)
+                    if (showQuickSettings)
                     {
                         // Quick settings is active, just quit that
-                        if (NULL != overlayMode->fnExitMode)
-                        {
-                            overlayMode->fnExitMode();
-                        }
-
-                        overlayMode = NULL;
+                        quickSettingsMode.fnExitMode();
+                        showQuickSettings = false;
                     }
                     else
                     {
                         // Quick settings not active, set it up
-                        overlayMode = &quickSettingsMode;
-
-                        // Init the quick settings mode
-                        if (overlayMode->fnEnterMode != NULL)
-                        {
-                            overlayMode->fnEnterMode();
-                        }
+                        quickSettingsMode.fnEnterMode();
+                        showQuickSettings = true;
                     }
 
                     // Reset the count
                     timePausePressed = 0;
                 }
                 else
                 {
-                    int16_t numPx = (tHeldUs * TFT_WIDTH) / PAUSE_TIME_US;
-                    fillDisplayArea(0, 0, numPx, 10, c333);
+                    int16_t numPx = (tHeldUs * (TFT_WIDTH - 60)) / PAUSE_TIME_US;
+                    fillDisplayArea(30, 0, 30 + numPx, 10, c333);
```

Comment:
You should `#define` these numbers in a header to ensure the width of the bar matches the width of the quick settings menu.

Reply:
It is consistent at least. You could mathematically figure out that `30` too.
```c
/// The width of the panel, centered in the screen
#define PANEL_W (TFT_WIDTH - 60)
```

Reply:
or do
```c
#define PANEL_MARGIN 30
#define PANEL_WIDTH (TFT_WIDTH - (2 * PANEL_MARGIN))
```